prometheus/promauto/auto.go

```diff
@@ -173,7 +173,9 @@ func NewCounter(opts prometheus.CounterOpts) prometheus.Counter {
 // prometheus.DefaultRegisterer. If the registration fails, NewCounterVec
 // panics.
 func NewCounterVec(opts prometheus.CounterOpts, labelNames []string) *prometheus.CounterVec {
-	return With(prometheus.DefaultRegisterer).NewCounterVec(opts, labelNames)
+	v := With(prometheus.DefaultRegisterer).NewCounterVec(opts, labelNames)
+	v.WithLabelValues(labelNames...)
```

Comment:
You are using the label _names_ as label _values_ here. That can't be right, can it?